File: leetcode/go/1860.incremental-memory-leak.go
func memLeak(memory1 int, memory2 int) []int {
    process := 1
    for process <= memory1 || process <= memory2 {
        if memory1 >= memory2 {
            memory1 -= process
        } else {
            memory2 -= process
        }
        process += 1
    }
    return [] int {process, memory1, memory2}
}

File: leetcode/rust-leetcode/src/1860.incremental-memory-leak.rs
impl Solution {
    pub fn mem_leak(mut memory1: i32, mut memory2: i32) -> Vec<i32> {
        let mut program = 1;
        while program <= memory1.max(memory2) {
            if memory1 >= memory2 {
                memory1 -= program;
            } else {
                memory2 -= program;
            }
            program += 1;
        }
        vec![program, memory1, memory2]
    }
}

